controllers/clustertopology_controller_compute.go

```diff
@@ -308,6 +332,101 @@ func computeCluster(current *clusterTopologyState, infrastructureCluster, contro
 	return cluster
 }
 
+// computeMachineDeploymentTopologyState computes the desired state for a MachineDeploymentTopology
+func computeMachineDeploymentTopologyState(class *clusterTopologyClass, current *clusterTopologyState, machineDeploymentTopology clusterv1.MachineDeploymentTopology, currentMachineDeployment *clusterv1.MachineDeployment) (machineDeploymentTopologyState, error) {
+	state := machineDeploymentTopologyState{}
+
+	className := machineDeploymentTopology.Class
+	machineDeploymentClass := getMachineDeploymentClass(class, className)
+
+	machineDeploymentTopologyClass, ok := class.machineDeployments[className]
+	if !ok {
+		return state, errors.Errorf("class name %s not found", className)
```

Comment:
I think some hints where we couldn't find it would be good (i.e. in the topology of the cluster + the cluster name)